overlord/fdestate/backend/reseal.go

```diff
@@ -60,7 +60,11 @@ func MockSecbootBuildPCRProtectionProfile(f func(modelParams []*secboot.SealKeyM
 	}
 }
 
-type StateUpdater func(role string, containerRole string, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte) error
+type FDEStateManager interface {
+	Update(role string, containerRole string, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte) error
+	Get(role string, containerRole string) (hasParameters bool, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte, err error)
+	Unlock() (relock func())
```

Comment:
```suggestion
	// Update updates the state information for keys used in a role and stored in the indicated container.
	Update(role string, containerRole string, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte) error
	// Get retrieves ...
	Get(role string, containerRole string) (hasParameters bool, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte, err error)
	// Unlock unlocks whatever 'state' the manager is keeping locked and returns an function to
	// take the state lock again.
	Unlock() (relock func())
```

on a side note, it looks like `bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte` could be passed around in a single struct? say ResealContextData or something along those lines. The only downside is that the struct would be defined in the backend and the package would need to be imported wherever we mock it somewhere.